packages/openstef-beam/src/openstef_beam/analysis/visualizations/windowed_metric_visualization.py

```diff
@@ -228,5 +266,131 @@ def create_by_target(
 
         return VisualizationOutput(name=self.name, figure=figure)
 
+    # averaging over all targets in a single group
+    @override
+    def create_by_run_and_target(
+        self,
+        reports: dict[RunName, list[ReportTuple]],
+    ) -> VisualizationOutput:
+        metric_name, quantile_or_global = self._get_metric_info()
+        plotter = WindowedMetricPlotter()
+
+        # Process each run and calculate averaged metrics across its targets
+        for run_name, target_reports in reports.items():
+            if not target_reports:
+                continue
+
+            # Average windowed metrics across all targets for this run
+            averaged_pairs = self._average_time_series_across_targets(
+                reports=target_reports,
+                metric_name=metric_name,
+                quantile_or_global=quantile_or_global,
+            )
+
+            # Skip if no averaged data points found for this run
+            if not averaged_pairs:
+                continue
```

Comment:
Instead of all the `continue` calls I would raise an error (as in the other methods in this class) like:
```suggestion
            if not averaged_pairs:
                    raise ValueError("No windowed metrics found for the specified window and metric across all groups.")
```
(If the `StrictExecutionCallback` is not added in a pipeline, this should not result in a fatal execution :))